lndr-backend/src/Lndr/Server.hs

```diff
@@ -228,19 +230,32 @@ verifySettlementsWithTxHash :: LndrHandler ()
 verifySettlementsWithTxHash = do
     (ServerState pool configTVar _) <- ask
     config <- liftIO $ readTVarIO configTVar
-    creditHashes <- liftIO $ withResource pool Db.settlementCreditsToVerify
-    mapM_ verifyIndividualRecord creditHashes
+    txHashes <- liftIO $ withResource pool Db.txHashesToVerify
+    creditsToVerify <- mapM (\txHash -> liftIO $ withResource pool $ Db.lookupCreditsByTxHash txHash) txHashes
+    mapM_ (\txGroupList -> verifyRecords txGroupList) creditsToVerify
 
 
-verifyIndividualRecord :: TransactionHash -> LndrHandler ()
-verifyIndividualRecord creditHash = do
+verifyRecords :: [BilateralCreditRecord] -> LndrHandler ()
+verifyRecords records = do
     (ServerState pool configTVar loggerSet) <- ask
-    recordM <- liftIO $ withResource pool $ Db.lookupCreditByHash creditHash
-    let recordNotFound = throwError $
-            err404 { errBody = "Credit hash does not refer to pending bilateral settlement record" }
-    bilateralCreditRecord <- maybe recordNotFound pure recordM
-    verifySettlementPayment bilateralCreditRecord
-    liftIO $ withResource pool $ Db.verifyCreditByHash creditHash
-    web3Result <- finalizeTransaction configTVar bilateralCreditRecord
-                        `catchError` (pure . T.pack . show)
+    
+    -- this should only take the creditor, debtor, credit hash, and settlement amount
+    let initialTxHash = txHash $ head records
+        initialRecord = creditRecord $ head records
+        initialSettlement = SettlementVerification (creditor initialRecord) (debtor initialRecord) initialTxHash (fromMaybe 0 $ settlementAmount initialRecord)
+        settlementRecord = foldr generateCompositeSettlement initialSettlement (tail records)
+    verifySettlementPayment settlementRecord
+    mapM_ (\(BilateralCreditRecord creditRecord _ _ _) -> liftIO $ withResource pool $ Db.verifyCreditByHash $ hash creditRecord) records
+    web3Result <- mapM (\bilateralCreditRecord -> finalizeTransaction configTVar bilateralCreditRecord) records
+                -- `catchError` (pure . T.pack . show)
     liftIO $ pushLogStrLn loggerSet . toLogStr . ("WEB3: " ++) . show $ web3Result
+
+
+generateCompositeSettlement :: BilateralCreditRecord -> SettlementVerification -> SettlementVerification
+generateCompositeSettlement (BilateralCreditRecord creditRecord _ _ txHash) (SettlementVerification creditorAddr debtorAddr txnHash settlementValue) =
+    let direction = if creditorAddr == creditor creditRecord then 1 else -1
+        newValue = direction * fromMaybe 0 (settlementAmount creditRecord) + settlementValue
+        newCreditor = if newValue > 0 then creditorAddr else debtorAddr
```

Comment:
This is unnecessary, as is the `foldr` above. Do not create this additional function and don't create an extra type `SettlementVerification`. I recommend passing creditor, debtor, txhash (which shouldn't potentially exist as `Maybe`, but must exist), and value to the `verifySettlementPayment` function.

You can quickly calculate who's owed in the body of `verifyRecords` by filtering the records lists by creditor debtors and running `sum` over the `settlementAmount`s.

Do you expect all of these Bilateral credit records to have an associated `settlementAmount`?